### PR TITLE
Issue #127: Fix post-install script

### PR DESCRIPTION
**Pull request by webflo**
This should fix the failing create-project command. Composer changed the behaviour in https://github.com/composer/composer/commit/55b0ed8c8be74af348d9fec6702da3efd50feb46

### CODE REVIEW

**Comment by greg-1-anderson** [comment]
I'm a little bit confused about the reference to the composer commit above. I just ran `composer create-project`, and it still works as well as it always did. Did Composer behavior really change at all here?

In any event, Composer changing behavior or not changing behavior doesn't really matter--I am just curious for my own edification. As far as this PR is concerned, switching the post-install script from a bash script to equivalent PHP code means that we are no longer dependent on `composer` being available in the user's $PATH at post-install time, so this seems like a good enhancement to me.

+1.

**Comment by webflo** [comment]
@greg-1-anderson You can spot the change in [create-project-command.test](https://github.com/composer/composer/commit/55b0ed8c8be74af348d9fec6702da3efd50feb46#diff-c1506ae77a3a87c8ff8479560f5181d8L10) easily.

File: scripts/composer/ScriptHandler.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+namespace DrupalProject\composer;
+
+use Composer\Script\Event;
+use Symfony\Component\Filesystem\Filesystem;
+
+class ScriptHandler {
+
+  protected static function getDrupalRoot($project_root) {
+    return $project_root .  '/web';
```
Comment (jibran) [inline scripts/composer/ScriptHandler.php:16]: I think we should get this directory(`/web`) form installer paths in composer.json.

File: scripts/composer/ScriptHandler.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    // Required for unit testing
+    foreach ($dirs as $dir) {
+      if (!$fs->exists($root . '/'. $dir)) {
+        $fs->mkdir($root . '/'. $dir);
+        $fs->touch($root . '/'. $dir . '/.gitkeep');
+      }
+    }
+
+    // Prepare the settings file for installation
```
Comment (jibran) [inline scripts/composer/ScriptHandler.php:44]: I think we should allow user to turn copying file on or off using some variable in `composer.json`. If you like the idea we can create a new issue for this.

File: scripts/composer/ScriptHandler.php (hunk 1 above)
Comment (webflo) [inline scripts/composer/ScriptHandler.php:16]: Yeah drupal-scaffold has this feature already. I think we could detect it via composer installers path or add something link "drupal-root-dir" to the composer.json. A own subclass of ScriptHandler should work as well ...

**Pull request merged**